### PR TITLE
check if default two qubit gate is supported by the device

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Solve https://github.com/CQCL/pytket-quantinuum/issues/457

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Comment by yao-cqc** [comment]
Can we instead check this in the `QuantinuumBackend` constructor and use the same error message in `set_compilation_config_target_2qb_gate`?

**Comment by cqc-melf** [comment]
> `set_compilation_config_target_2qb_gate`

`set_compilation_config_target_2qb_gate` already has a check, so I think that should be fine?

Unfortunately I can't use _gate_set in the constructor because that calls the API and breaks quiet a few things for backends constructed with the offlineapi.

Happy to find a workaround for that if you think that is important.

**Comment by cqc-alec** [comment]
It does feel wrong to set this config value inside rebase_pass(). Can we change the `_default_2q_gate()` function instead?

**Comment by cqc-melf** [comment]
> By the way, `deadhead` and `brathead` both now accept the `ZZPhase` gate. So maybe we don't need this anymore, as `ZZPhase` is available on all devices?

I think we discussed this in the last sprint planing that we still want to have this check additionally to make this future proof.  

If you think this is not required, then we can close this.

**Comment by cqc-melf** [comment]
> > > `set_compilation_config_target_2qb_gate`
> > 
> > 
> > `set_compilation_config_target_2qb_gate` already has a check, so I think that should be fine?
> > Unfortunately I can't use _gate_set in the constructor because that calls the API and breaks quiet a few things for backends constructed with the offlineapi.
> > Happy to find a workaround for that if you think that is important.
> 
> I think `self._gate_set` won't make any API call if the backend is constructed with the offline API. Maybe I misunderstood the problem?

I think I was wrong with the problem detection, the actual issue is that we have tests are not expected to make API calls that would make this calls with the changes in the constructor.

I have now gone the way and updated all the affected testcases.

**Review by cqc-alec — COMMENTED** [review]
I'm just a little worried that requiring login when constructing `QuantinuumBackend` may break some workflows; perhaps we should check this with nexus?

**Comment by cqc-melf** [comment]
> I'm just a little worried that requiring login when constructing `QuantinuumBackend` may break some workflows; perhaps we should check this with nexus?

I checked with them, this is not a concern.

**Comment by cqc-melf** [comment]
I will take a look at the failed notebook and update that. Will ask for a review after that is updated.

**Comment by cqc-alec** [comment]
> I will take a look at the failed notebook and update that. Will ask for a review after that is updated.

This is probably #474 . If so it needs an update from the US side.

**Review by cqc-alec — DISMISSED** [review]

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 1]
```diff
@@ -1038,7 +1038,7 @@ def process_circuits(
                                 "max_classical_register_width", 32
```
Comment (cqc-alec) [inline pytket/extensions/quantinuum/backends/quantinuum.py]: If we want to change the default we should change it here too, but I'm not sure if we want to do this.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**